### ISSUE TITLE
[FEATURE]: pass `AbstractSampler.y` to `stratify` in sklearn `train_test_split` function in `Random` sampler if provided

### ISSUE THREAD

**Issue by lukethomrichardson** [JacksonBurns/astartes#186]
**Use-cases/examples of this new feature**

Stratification of random splits is a common tool in ML when class labels are imbalanced.

**Desired solution/workflow**

Pass  `AbstractSampler.y` attribute to the `stratify` kwarg argument of the `sk_train_test_split` function call in the `Random` sampler definition.

**Comment by JacksonBurns** [comment]
Hi @lukethomrichardson thanks for the suggestion. Stratification is well defined for the random sampler, but not so much for the others (e.g. kennard stone). For the time being, we will leave this as-is, and suggest directing calling scikit-learn's random sampler if stratification is needed.
